llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -616,34 +632,15 @@ void AIEPostRASchedStrategy::leaveRegion(const SUnit &ExitSU) {
   assert(BS.getCurrentRegion().Bundles.empty());
   BS.addBundles(TopBundles);
   BS.addBundles(BotBundles);
+  if (!ReAssignMultiSlotInstr)
+    materializeMultiOpcodeInstrs();
```

Comment:
Any particular reason to "move" the lines down here? What if we do not enter `if (IsBottomRegion) { ... }`?

Reply:
The following code block is outside the `if (IsBottomRegion) { ... }`
```
  if (!ReAssignMultiSlotInstr)
    materializeMultiOpcodeInstrs();
```